diesel_codegen_syntex/src/update.rs

```diff
@@ -3,80 +3,86 @@ use syntax::attr::AttrMetaMethods;
 use syntax::codemap::Span;
 use syntax::ext::base::{Annotatable, ExtCtxt};
 use syntax::ptr::P;
-use syntax::parse::token::{InternedString, str_to_ident};
-use syntax::tokenstream::TokenTree;
 
 use model::Model;
-use util::lifetime_list_tokens;
+use util::{lifetime_list_tokens, str_value_of_attr_with_name};
 
-pub fn expand_changeset_for(
+pub fn expand_derive_as_changeset(
     cx: &mut ExtCtxt,
     span: Span,
     meta_item: &MetaItem,
     annotatable: &Annotatable,
     push: &mut FnMut(Annotatable),
 ) {
     if let Some(model) = Model::from_annotable(cx, span, annotatable) {
-        let options = changeset_options(cx, meta_item).unwrap();
+        let options = changeset_options(cx, span, annotatable.attrs()).unwrap();
         push(Annotatable::Item(changeset_impl(cx, span, &options, &model).unwrap()));
     } else {
         cx.span_err(meta_item.span,
-            "`changeset_for` may only be applied to enums and structs");
+            "`#[derive(AsChangeset)]` may only be applied to enums and structs");
     }
 }
 
 struct ChangesetOptions {
     table_name: ast::Ident,
-    treat_none_as_null: Vec<TokenTree>,
+    treat_none_as_null: bool,
 }
 
-fn changeset_options(cx: &mut ExtCtxt, meta_item: &MetaItem) -> Result<ChangesetOptions, ()> {
-    match meta_item.node {
-        MetaItemKind::List(_, ref meta_items) => {
-            let table_name = try!(table_name(cx, &meta_items[0]));
-            let treat_none_as_null = try!(boolean_option(cx, &meta_items[1..], "treat_none_as_null"));
-            Ok(ChangesetOptions {
-                table_name: str_to_ident(&table_name),
-                treat_none_as_null: treat_none_as_null,
-            })
+fn changeset_options(
+    cx: &mut ExtCtxt,
+    span: Span,
+    attributes: &[ast::Attribute]
+) -> Result<ChangesetOptions, ()> {
+    let changeset_options_attr = attributes.iter().find(|a| a.check_name("changeset_options"));
+    let treat_none_as_null = try!(changeset_options_attr
+        .map(|a| extract_treat_none_as_null(cx, a))
+        .unwrap_or(Ok(false)));
```

Comment:
This will probably clarify:

```rust
let x: Option<&ast::Attribute> = x;
let y: Option<Result<bool, ()>> = y.map(|a| extract_treat_none_as_null(cx, a));
let z: Result<bool, ()> = z.unwrap_or(Ok(false));
let w: bool = try!(z);
```